iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/tsfile/timeindex/FileTimeIndexCacheRecorder.java

```diff
@@ -193,10 +193,11 @@ public void close() throws IOException {
   }
 
   public void removeFileTimeIndexCache(int dataRegionId) {
-    for (FileTimeIndexCacheWriter writer : writerMap.values()) {
+    FileTimeIndexCacheWriter writer = writerMap.get(dataRegionId);
```

Comment:
We need to maintain the writerMap when there are region migrations and drop database

Reply:
fixed